### DISCUSSION TITLE
CDN caching for SSR functions affected by auth

### DISCUSSION THREAD

**Discussion by bwesner** [vercel/vercel#5343] [Help]
I have a situation where all of my SSR functions return different content depending on whether or not the user is logged in. Specifically, there is a nav bar on all pages that has different content for a logged in user. Ideally, I want to cache the non-auth version of the page, which is the same for all non-auth'd users. And if a user has auth, I want them to always hit the server instead of hitting the cache.

I try to accomplish this by adding the following header to all pages through `next.config.js`:

```
{
    key: 'Cache-Control',
    value: 's-maxage=30, stale-while-revalidate=60'
}
```

Then, in the SSR logic, I do something like:

```
if (token) { // token is from cookies and indicates whether or not the user is auth'd
    res.setHeader('Cache-Control', 'no-store');
}
```

However, this doesn't quite work as I expect. It seems that, when auth'd, sometimes the SSR request to one of the pages will return the non-auth'd version of the page. Does anyone know how to accomplish what I am trying to do without moving the auth-based stuff into a client-side call?

**Comment by regenrek** [comment]
Hi @bwesner 

did you ever solve this?

**Reply by bwesner** [reply]
Nope. Ended up completely refactoring my app so that the server would return the non-auth'd version of the page 100% of the time, and then the client makes a second request for user info right after the initial page load. This way, I can confidently cache the SSR'd page. To help reduce any UI switching an auth'd user might see, on the client I keep track of whether or not a user is logged in, so that after the initial page load on the SSR'd pages in question, I can immediately optimistically switch the UI to at least a semblance of the auth'd version of the UI, instead of having to wait for that user info call to make the UI switch.
